### PR TITLE
fix 'The href of a link processsed incorrectly in an iframe without src if it is set from the main window' (close #620)

### LINKED ISSUE #620: The href of a link processsed incorrectly in an iframe without src if it is set from the main window
There is an iframe

``` html
<iframe id="iframe"></iframe>
```

If I create a link inside the iframe in the following way:

``` js
// This code executes in the context of the top window
var iframe = document.getElementById('iframe');
var body = iframe.contentDocument.body;

body.innerHTML = '<a href="somePage.html">link</a>';
```

the link has processed href without the `iframe` flag. As I result, when I click on it, the new page in the iframe has `task.js` script instead of `iframe-task.js`.

If I do it in the other way:

``` js
// This code executes in the context of the top window
 var iframeDocument = document.getElementById('iframe').contentWindow.document;

var link = iframeDocument.createElement('a');

link.textContent = 'link';
link.href = 'somePage.html';
iframeDocument.body.appendChild(link);
```

it works ok in chrome and ie, but still broken in firefox.

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev

### CODE REVIEW

**Comment by LavrovArtem** [comment]
ping

**Comment by miherlosev** [comment]
\r-

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 1]
```diff
@@ -810,6 +810,17 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
         return window[ORIGINAL_WINDOW_ON_ERROR_HANDLER_KEY];
     }
 
+    static _canAccessorsRedirect (window, instruction, owner) {
+        try {
+            return owner && owner[INTERNAL_PROPS.processedContext] &&
+                   owner[INTERNAL_PROPS.processedContext] !== window &&
```
Comment (churkin) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Why `window`? Maybe `this.window`?

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 2]
```diff
@@ -837,6 +848,9 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
                 if (typeUtils.isNullOrUndefined(owner))
                     PropertyAccessorsInstrumentation._error(`Cannot set property '${propName}' of ${typeUtils.inaccessibleTypeToStr(owner)}`);
 
+                if (PropertyAccessorsInstrumentation._canAccessorsRedirect(window, INSTRUCTION.setProperty, owner))
```
Comment (churkin) [inline src/client/sandbox/code-instrumentation/properties/index.js]: `_canAccessorsRedirect` is a bad name

File: src/client/sandbox/code-instrumentation/properties/index.js (hunk 1 above)
Comment (LavrovArtem) [inline src/client/sandbox/code-instrumentation/properties/index.js]: ` static _canAccessorsRedirect (window, instruction, owner) {`

Comment (churkin) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Oh, I see now

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 3]
```diff
@@ -837,6 +848,11 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
                 if (typeUtils.isNullOrUndefined(owner))
                     PropertyAccessorsInstrumentation._error(`Cannot set property '${propName}' of ${typeUtils.inaccessibleTypeToStr(owner)}`);
 
+                var ownerSetPropertyInstruction = PropertyAccessorsInstrumentation._getSetPropertyInstructionByOwner(owner, window);
```
Comment (miherlosev) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Do we need to do this for `INSTRUCTION.getProperty`?

Comment (LavrovArtem) [inline src/client/sandbox/code-instrumentation/properties/index.js]: I investigated getters of html elements and found out that no one of them is independent of current window.

Comment (miherlosev) [inline src/client/sandbox/code-instrumentation/properties/index.js]: Ok

**Pull request merged**